homeassistant/components/pylontech_us/__init__.py

```diff
@@ -0,0 +1,96 @@
+"""The pylontech_us integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from aioesphomeapi import Any
+import async_timeout
+from pylontech import PylontechStack
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+
+class PylontechCoordinator(DataUpdateCoordinator):
+    """Coordinator class to collect data from battery."""
+
+    def __init__(
+        self, hass: HomeAssistant, port: str, baud: int, battery_count: int
+    ) -> None:
+        """Pylontech setup."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=SCAN_INTERVAL,
+        )
+        self._stack = PylontechStack(
+            device=port, baud=baud, manualBattcountLimit=battery_count
+        )
+        self._last_result = self._stack.update()
```

Comment:
Presumably this does blocking I/O here and should not run in the constructor. If the idea is to get a value right away the i think `async_config_entry_first_refresh` is what you may want.

Reply:
Thanks for the hint. - Moved.